src/BlockTypes/ProductReviews.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Automattic\WooCommerce\Blocks\BlockTypes;
+
+/**
+ * ProductReviews class.
+ */
+class ProductReviews extends AbstractBlock {
+	/**
+	 * Block name.
+	 *
+	 * @var string
+	 */
+	protected $block_name = 'product-reviews';
+
+	/**
+	 * It isn't necessary register block assets because it is a server side block.
+	 */
+	protected function register_block_type_assets() {
+		return null;
+	}
+
+	/**
+	 * Render the block.
+	 *
+	 * @param array    $attributes Block attributes.
+	 * @param string   $content Block content.
+	 * @param WP_Block $block Block instance.
+	 *
+	 * @return string Rendered block output.
+	 */
+	protected function render( $attributes, $content, $block ) {
+		$post_id = $block->context['postId'];
+		$old_global_product = $GLOBALS['product'];
+		$old_global_query = $GLOBALS['wp_query'];
+
+		$new_query = new \WP_Query(
+			array(
+				'feed'         => 1,
+				'p'            => $post_id,
+				'post_type'    => 'product',
+				'withcomments' => 1,
+			)
+		);
+
+		// Ovewriting the globals, because we need it in the template.
+		$GLOBALS['product'] = wc_get_product( $post_id );
+		$GLOBALS['wp_query'] = $new_query; // phpcs:ignore
+
+		wc_get_template( 'single-product-reviews.php' );
+
+		$reviews = ob_get_clean();
+
+		// Resetting the globals to what it was.
+		$GLOBALS['product'] = $old_global_product;
+		$GLOBALS['wp_query'] = $old_global_query; // phpcs:ignore
```

Comment:
What do you think about following this route?

```suggestion
        ob_start();

		rewind_posts();
		while ( have_posts() ) {
			the_post();
			comments_template();
		}

		$reviews = ob_get_clean();
```

By invoking `comments_template();` instead of `wc_get_template( 'single-product-reviews.php' );`, we:

- Ensure all necessary variables are in place, so there's no need to overwrite/reset globals
- Defer the loading of the `'single-product-reviews.php'` template to the core of Woo: the correct template will still be loaded as soon as the `comment_template` filter is triggered.

Also, we can benefit from `rewind_posts()` for updating the internal pointer for the posts and preventing the conflicts with the Post Content block that we were discussing [over here](https://github.com/woocommerce/woocommerce-blocks/discussions/8792#discussioncomment-5387577).

Reply:
I'm totally fine with proceeding with this direction, especially because it is consistent with what Gutenberg does. The discussion could stay open in case anyone wants to share their opinion (Nik from Gutenberg, for instance, was also suggesting the second point), but I feel the need to elaborate any further, personally: you make good points and, at the end of the day, these are implementation details that we can easily change without causing breaking changes if we ever feel the need to, so I'm totally happy to proceed with this, if you are satisfied as well :)